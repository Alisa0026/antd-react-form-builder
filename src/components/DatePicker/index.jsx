import React from 'react'
import _ from 'lodash'
import moment from 'moment'
import BasicItem from '../BasicItem'
import DatePicker from 'antd/lib/date-picker'
import FormItemComponentDecorator from '../../decorator/FormItemComponent'

function component(BasicItemComponent){
  @FormItemComponentDecorator
  class FDatePicker extends React.Component {

    render(){
      let { ...other } = this.props;

      other.targetComponent = DatePicker;
      if(_.isString(other.value)){
        other.value = new moment(other.value,other.format || "YYYY-MM-DD")
      }
      other.type = "datepicker";
      this.propsAdapter(other);
      return (
        <BasicItemComponent { ...other }/>
      )
    }
    
  }
  return FDatePicker;
}

export default component(BasicItem) 




