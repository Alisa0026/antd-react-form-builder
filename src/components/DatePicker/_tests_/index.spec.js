import React from 'react'
import { shallow,mount,render } from 'enzyme'
import sinon from 'sinon'
import moment from 'moment'
import LocaleProvider from 'antd/lib/locale-provider'
import { 
  testCommonProps, 
  getAntdFormComponentMount,
  getAntdFormItemMount,
} from '../../../../tests/common'
import DatePicker from '../index'

describe('DatePicker',function(){
  sinon.stub(console,"warn");
  it('renders correctly', () => {
    const wrapper = render(
      <DatePicker 
        required
        className="test"
        label="name"
        name="test"
        value={
          moment("2016-04-14","YYYY-MM-DD").utc()
        }
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  testCommonProps(DatePicker); 
  it("stirng value should be rendered correctly",function(){
    const wrapper = mount(
      <DatePicker 
        required
        className="test"
        label="name"
        name="test"
        value="2016-04-14"
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(new moment("2016-04-14","YYYY-MM-DD"));
  })
  it("moment value should be rendered correctly",function(){
    const wrapper = mount(
      <DatePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ moment("2016-04-14") }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(antdInputMount.props().value).toEqual(moment("2016-04-14"));
  })
  it("should have hidden <Input /> ",function(){
    var value = new moment("2016-04-14");
    const wrapper = mount(
      <DatePicker 
        required
        className="test"
        label="name"
        name="test"
        value={ value }
      />
    );
    const antdInputMount = getAntdFormComponentMount(wrapper,1);
    const antdFormItemMount = getAntdFormItemMount(wrapper);
    //console.log(antdInputMount.pr)
    expect(wrapper.childAt(1).props().value).toEqual(Math.floor(+value / 1000));
    expect(wrapper.childAt(1).props().name).toEqual('test');
  })
})

