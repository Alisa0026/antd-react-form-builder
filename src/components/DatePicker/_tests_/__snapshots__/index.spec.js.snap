exports[`DatePicker renders correctly 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {},
        "children": Array [
          Object {
            "attribs": Object {
              "class": "ant-row ant-form-item",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "ant-form-item-label",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "ant-form-item-required",
                      "title": "name",
                    },
                    "children": Array [
                      Object {
                        "data": "name",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "label",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": Object {
                  "attribs": Object {
                    "class": "ant-form-item-control-wrapper",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-form-item-control has-success",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "ant-calendar-picker",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {},
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-input ant-input",
                                    "placeholder": "请选择日期",
                                    "readonly": "",
                                    "value": "2016-04-14",
                                  },
                                  "children": Array [],
                                  "name": "input",
                                  "next": Object {
                                    "attribs": Object {
                                      "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                    },
                                    "children": Array [],
                                    "name": "i",
                                    "next": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-icon",
                                      },
                                      "children": Array [],
                                      "name": "span",
                                      "next": null,
                                      "parent": [Circular],
                                      "prev": [Circular],
                                      "type": "tag",
                                    },
                                    "parent": [Circular],
                                    "prev": [Circular],
                                    "type": "tag",
                                  },
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                },
                                Object {
                                  "attribs": Object {
                                    "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                  },
                                  "children": Array [],
                                  "name": "i",
                                  "next": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-icon",
                                    },
                                    "children": Array [],
                                    "name": "span",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": [Circular],
                                    "type": "tag",
                                  },
                                  "parent": [Circular],
                                  "prev": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-input ant-input",
                                      "placeholder": "请选择日期",
                                      "readonly": "",
                                      "value": "2016-04-14",
                                    },
                                    "children": Array [],
                                    "name": "input",
                                    "next": [Circular],
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                  },
                                  "type": "tag",
                                },
                                Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-icon",
                                  },
                                  "children": Array [],
                                  "name": "span",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": Object {
                                    "attribs": Object {
                                      "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                    },
                                    "children": Array [],
                                    "name": "i",
                                    "next": [Circular],
                                    "parent": [Circular],
                                    "prev": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-input ant-input",
                                        "placeholder": "请选择日期",
                                        "readonly": "",
                                        "value": "2016-04-14",
                                      },
                                      "children": Array [],
                                      "name": "input",
                                      "next": [Circular],
                                      "parent": [Circular],
                                      "prev": null,
                                      "type": "tag",
                                    },
                                    "type": "tag",
                                  },
                                  "type": "tag",
                                },
                              ],
                              "name": "span",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                            },
                          ],
                          "name": "span",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                      ],
                      "name": "div",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
              },
              Object {
                "attribs": Object {
                  "class": "ant-form-item-control-wrapper",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "class": "ant-form-item-control has-success",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "ant-calendar-picker",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {},
                            "children": Array [
                              Object {
                                "attribs": Object {
                                  "class": "ant-calendar-picker-input ant-input",
                                  "placeholder": "请选择日期",
                                  "readonly": "",
                                  "value": "2016-04-14",
                                },
                                "children": Array [],
                                "name": "input",
                                "next": Object {
                                  "attribs": Object {
                                    "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                  },
                                  "children": Array [],
                                  "name": "i",
                                  "next": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-icon",
                                    },
                                    "children": Array [],
                                    "name": "span",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": [Circular],
                                    "type": "tag",
                                  },
                                  "parent": [Circular],
                                  "prev": [Circular],
                                  "type": "tag",
                                },
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                              },
                              Object {
                                "attribs": Object {
                                  "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                },
                                "children": Array [],
                                "name": "i",
                                "next": Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-icon",
                                  },
                                  "children": Array [],
                                  "name": "span",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": [Circular],
                                  "type": "tag",
                                },
                                "parent": [Circular],
                                "prev": Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-input ant-input",
                                    "placeholder": "请选择日期",
                                    "readonly": "",
                                    "value": "2016-04-14",
                                  },
                                  "children": Array [],
                                  "name": "input",
                                  "next": [Circular],
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                },
                                "type": "tag",
                              },
                              Object {
                                "attribs": Object {
                                  "class": "ant-calendar-picker-icon",
                                },
                                "children": Array [],
                                "name": "span",
                                "next": null,
                                "parent": [Circular],
                                "prev": Object {
                                  "attribs": Object {
                                    "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                  },
                                  "children": Array [],
                                  "name": "i",
                                  "next": [Circular],
                                  "parent": [Circular],
                                  "prev": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-input ant-input",
                                      "placeholder": "请选择日期",
                                      "readonly": "",
                                      "value": "2016-04-14",
                                    },
                                    "children": Array [],
                                    "name": "input",
                                    "next": [Circular],
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                  },
                                  "type": "tag",
                                },
                                "type": "tag",
                              },
                            ],
                            "name": "span",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                          },
                        ],
                        "name": "span",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                ],
                "name": "div",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "ant-form-item-label",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-form-item-required",
                        "title": "name",
                      },
                      "children": Array [
                        Object {
                          "data": "name",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                "type": "tag",
              },
            ],
            "name": "div",
            "next": Object {
              "attribs": Object {
                "class": "ant-input",
                "name": "test",
                "type": "hidden",
                "value": "1460563200",
              },
              "children": Array [],
              "name": "input",
              "next": null,
              "parent": [Circular],
              "prev": [Circular],
              "type": "tag",
            },
            "parent": [Circular],
            "prev": null,
            "type": "tag",
          },
          Object {
            "attribs": Object {
              "class": "ant-input",
              "name": "test",
              "type": "hidden",
              "value": "1460563200",
            },
            "children": Array [],
            "name": "input",
            "next": null,
            "parent": [Circular],
            "prev": Object {
              "attribs": Object {
                "class": "ant-row ant-form-item",
              },
              "children": Array [
                Object {
                  "attribs": Object {
                    "class": "ant-form-item-label",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-form-item-required",
                        "title": "name",
                      },
                      "children": Array [
                        Object {
                          "data": "name",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "label",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": Object {
                    "attribs": Object {
                      "class": "ant-form-item-control-wrapper",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "ant-form-item-control has-success",
                        },
                        "children": Array [
                          Object {
                            "attribs": Object {
                              "class": "ant-calendar-picker",
                            },
                            "children": Array [
                              Object {
                                "attribs": Object {},
                                "children": Array [
                                  Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-input ant-input",
                                      "placeholder": "请选择日期",
                                      "readonly": "",
                                      "value": "2016-04-14",
                                    },
                                    "children": Array [],
                                    "name": "input",
                                    "next": Object {
                                      "attribs": Object {
                                        "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                      },
                                      "children": Array [],
                                      "name": "i",
                                      "next": Object {
                                        "attribs": Object {
                                          "class": "ant-calendar-picker-icon",
                                        },
                                        "children": Array [],
                                        "name": "span",
                                        "next": null,
                                        "parent": [Circular],
                                        "prev": [Circular],
                                        "type": "tag",
                                      },
                                      "parent": [Circular],
                                      "prev": [Circular],
                                      "type": "tag",
                                    },
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                  },
                                  Object {
                                    "attribs": Object {
                                      "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                    },
                                    "children": Array [],
                                    "name": "i",
                                    "next": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-icon",
                                      },
                                      "children": Array [],
                                      "name": "span",
                                      "next": null,
                                      "parent": [Circular],
                                      "prev": [Circular],
                                      "type": "tag",
                                    },
                                    "parent": [Circular],
                                    "prev": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-input ant-input",
                                        "placeholder": "请选择日期",
                                        "readonly": "",
                                        "value": "2016-04-14",
                                      },
                                      "children": Array [],
                                      "name": "input",
                                      "next": [Circular],
                                      "parent": [Circular],
                                      "prev": null,
                                      "type": "tag",
                                    },
                                    "type": "tag",
                                  },
                                  Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-icon",
                                    },
                                    "children": Array [],
                                    "name": "span",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": Object {
                                      "attribs": Object {
                                        "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                      },
                                      "children": Array [],
                                      "name": "i",
                                      "next": [Circular],
                                      "parent": [Circular],
                                      "prev": Object {
                                        "attribs": Object {
                                          "class": "ant-calendar-picker-input ant-input",
                                          "placeholder": "请选择日期",
                                          "readonly": "",
                                          "value": "2016-04-14",
                                        },
                                        "children": Array [],
                                        "name": "input",
                                        "next": [Circular],
                                        "parent": [Circular],
                                        "prev": null,
                                        "type": "tag",
                                      },
                                      "type": "tag",
                                    },
                                    "type": "tag",
                                  },
                                ],
                                "name": "span",
                                "next": null,
                                "parent": [Circular],
                                "prev": null,
                                "type": "tag",
                              },
                            ],
                            "name": "span",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "tag",
                          },
                        ],
                        "name": "div",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": null,
                    "parent": [Circular],
                    "prev": [Circular],
                    "type": "tag",
                  },
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                },
                Object {
                  "attribs": Object {
                    "class": "ant-form-item-control-wrapper",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "class": "ant-form-item-control has-success",
                      },
                      "children": Array [
                        Object {
                          "attribs": Object {
                            "class": "ant-calendar-picker",
                          },
                          "children": Array [
                            Object {
                              "attribs": Object {},
                              "children": Array [
                                Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-input ant-input",
                                    "placeholder": "请选择日期",
                                    "readonly": "",
                                    "value": "2016-04-14",
                                  },
                                  "children": Array [],
                                  "name": "input",
                                  "next": Object {
                                    "attribs": Object {
                                      "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                    },
                                    "children": Array [],
                                    "name": "i",
                                    "next": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-icon",
                                      },
                                      "children": Array [],
                                      "name": "span",
                                      "next": null,
                                      "parent": [Circular],
                                      "prev": [Circular],
                                      "type": "tag",
                                    },
                                    "parent": [Circular],
                                    "prev": [Circular],
                                    "type": "tag",
                                  },
                                  "parent": [Circular],
                                  "prev": null,
                                  "type": "tag",
                                },
                                Object {
                                  "attribs": Object {
                                    "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                  },
                                  "children": Array [],
                                  "name": "i",
                                  "next": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-icon",
                                    },
                                    "children": Array [],
                                    "name": "span",
                                    "next": null,
                                    "parent": [Circular],
                                    "prev": [Circular],
                                    "type": "tag",
                                  },
                                  "parent": [Circular],
                                  "prev": Object {
                                    "attribs": Object {
                                      "class": "ant-calendar-picker-input ant-input",
                                      "placeholder": "请选择日期",
                                      "readonly": "",
                                      "value": "2016-04-14",
                                    },
                                    "children": Array [],
                                    "name": "input",
                                    "next": [Circular],
                                    "parent": [Circular],
                                    "prev": null,
                                    "type": "tag",
                                  },
                                  "type": "tag",
                                },
                                Object {
                                  "attribs": Object {
                                    "class": "ant-calendar-picker-icon",
                                  },
                                  "children": Array [],
                                  "name": "span",
                                  "next": null,
                                  "parent": [Circular],
                                  "prev": Object {
                                    "attribs": Object {
                                      "class": "anticon anticon-cross-circle ant-calendar-picker-clear",
                                    },
                                    "children": Array [],
                                    "name": "i",
                                    "next": [Circular],
                                    "parent": [Circular],
                                    "prev": Object {
                                      "attribs": Object {
                                        "class": "ant-calendar-picker-input ant-input",
                                        "placeholder": "请选择日期",
                                        "readonly": "",
                                        "value": "2016-04-14",
                                      },
                                      "children": Array [],
                                      "name": "input",
                                      "next": [Circular],
                                      "parent": [Circular],
                                      "prev": null,
                                      "type": "tag",
                                    },
                                    "type": "tag",
                                  },
                                  "type": "tag",
                                },
                              ],
                              "name": "span",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "tag",
                            },
                          ],
                          "name": "span",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                        },
                      ],
                      "name": "div",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                    },
                  ],
                  "name": "div",
                  "next": null,
                  "parent": [Circular],
                  "prev": Object {
                    "attribs": Object {
                      "class": "ant-form-item-label",
                    },
                    "children": Array [
                      Object {
                        "attribs": Object {
                          "class": "ant-form-item-required",
                          "title": "name",
                        },
                        "children": Array [
                          Object {
                            "data": "name",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "label",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                      },
                    ],
                    "name": "div",
                    "next": [Circular],
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                  },
                  "type": "tag",
                },
              ],
              "name": "div",
              "next": [Circular],
              "parent": [Circular],
              "prev": null,
              "type": "tag",
            },
            "type": "tag",
          },
        ],
        "name": "span",
        "next": null,
        "parent": null,
        "prev": null,
        "root": [Circular],
        "type": "tag",
      },
    ],
    "name": "root",
    "next": null,
    "parent": null,
    "prev": null,
    "type": "root",
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xmlMode": false,
  },
}
`;
